protos/flyteidl/service/plugin_system.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+package flyteidl.service;
+
+option go_package = "github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/service";
+import "flyteidl/core/literals.proto";
+import "flyteidl/core/tasks.proto";
+import "flyteidl/core/interface.proto";
+
+// BackendPluginService defines an RPC Service that allows propeller to send the request to the backend plugin server.
+service BackendPluginService {
+  // Send a task create request to the backend plugin server.
+  rpc CreateTask (TaskCreateRequest) returns (TaskCreateResponse){};
+  // Get job status.
+  rpc GetTask (TaskGetRequest) returns (TaskGetResponse){};
+  // Delete the task resource.
+  rpc DeleteTask (TaskDeleteRequest) returns (TaskDeleteResponse){};
+}
+
+// The state of the execution is used to control its visibility in the UI/CLI.
+enum State {
+  FAILED = 0;
+  PENDING = 1;
+  RUNNING = 2;
+  SUCCEEDED = 3;
+}
+
+// Represents a request structure to create task.
+message TaskCreateRequest {
+  // The inputs required to start the execution. All required inputs must be
+  // included in this map. If not required and not provided, defaults apply.
+  // +optional
+  core.LiteralMap inputs = 1;
+  // Template of the task that encapsulates all the metadata of the task.
+  core.TaskTemplate template = 2;
+  // Prefix for where offloaded data from user workflows will be written.
+  string output_prefix = 3;
+}
+
+// Represents a create response structure.
+message TaskCreateResponse {
```

Comment:
Is this a `oneof`? is there an instance where a `job_id` and `message` would be returned? Also can we be more descriptive of the message? ex. `error_message`

Reply:
updated it to `oneof`, thanks